### PR TITLE
[e2e]: CPU hotplug e2e fixes

### PR DESCRIPTION
**Pull request by enp0s3**
Signed-off-by: enp0s3 <ibezukh@redhat.com>

**What this PR does / why we need it**:
* The `countDomCPUs` helper function wasn't asserting with offset in some cases.
* In the dedicated CPU scenario there are two migrations that happen sequentially. The test was looking at the wrong migration object, since the object doesn't have any metadata about its originator, be it the hotplug mechanism or a manual trigger.

Fixes #
https://storage.googleapis.com/kubevirt-prow/reports/flakefinder/kubevirt/kubevirt/flakefinder-2023-06-26-168h.html#row3

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: tests/hotplug/cpu.go [hunk 1]
```diff
@@ -235,6 +235,10 @@ var _ = Describe("[sig-compute][Serial]CPU Hotplug", decorators.SigCompute, deco
 
 			tests.ExpectMigrationSuccess(virtClient, migration, tests.MigrationWaitTime)
 
+			By("delete the migration object from the namespace")
```
Comment (xpivarc) [inline tests/hotplug/cpu.go:238]: Wouldn't it be better to fix the loop below which should pick the correct migration?

Comment (enp0s3) [reply tests/hotplug/cpu.go:238]: @xpivarc the problem is that we don't annotate/label migrations that were caused by hotplug. I can label the manual migration and filter it.

Comment (xpivarc) [reply tests/hotplug/cpu.go:238]: Above you assert for the first migration to succeed. So there should be one migration that is not finalized.

Comment (enp0s3) [reply tests/hotplug/cpu.go:238]: @xpivarc but it could be flaky, it can finish before I fire the API call

Comment (xpivarc) [reply tests/hotplug/cpu.go:238]: If both are going to be finished that's okay as well.
`tests.ExpectMigrationSuccess(virtClient, migration, tests.MigrationWaitTime)` will succeed and everything else should be fine. Am I missing anything?

**Comment by enp0s3** [comment]
@xpivarc Lets wait for #9988 to get in. Then I will improve the `BeforeEach` as well (won't need to wait for KV sync)

**Comment by enp0s3** [comment]
@xpivarc PTAL

File: tests/hotplug/cpu.go [hunk 2]
```diff
@@ -52,7 +48,13 @@ var _ = Describe("[sig-compute][Serial]CPU Hotplug", decorators.SigCompute, deco
...
 		}
 		patchWorkloadUpdateMethod(originalKv.Name, virtClient, updateStrategy)
-		testsuite.EnsureKubevirtReadyWithTimeout(kubevirtReadyTimeoutSeconds)
+
+		currentKv := util2.GetCurrentKv(virtClient)
+		tests.WaitForConfigToBePropagatedToComponent(
+			"kubevirt.io=virt-controller",
+			currentKv.ResourceVersion,
+			tests.ExpectResourceVersionToBeLessEqualThanConfigVersion,
+			60*time.Second)
```
Comment (jean-edouard) [inline tests/hotplug/cpu.go:57]:
```suggestion
			time.Minute)
```
Will 1 minute be enough here?

Comment (enp0s3) [reply tests/hotplug/cpu.go]: @jean-edouard Yeah its pretty much time, the client cache on virt-controller side should receive the updated object by then, if IMO then we have some issue.

Comment (jean-edouard) [reply tests/hotplug/cpu.go]: /approve

**Review by dhiller — APPROVED** [review]

**Pull request merged**